refactor(elixir): session handshake and worker option names

davidag opened the pull request:
<!-- Thank you for sending a pull request :heart: -->

## Current Behavior

NA
<!-- Please describe the current behavior of the code before the changes in this pull request is applied. -->

## Proposed Changes

Apply a variable name refactor as suggested in https://github.com/build-trust/ockam/issues/3653

I'm not familiar with the code, but I'd like to make sure that you want to rename `worker_mod` to `data_mod`. I was wondering if `data_worker_mod` could be more meaningful. What do you think @hairyhum ?

<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

<!-- Looking forward to merging your contribution!! -->

davidag committed:
refactor(elixir): session handshake and worker option names

Naming was inconsistent because the handshake option refers to a module
and the worker module is the data worker module.

Spawner does not have a handshake and data stages, so we can refer to
its worker mod just as `worker_mod`, and to its options as
`worker_options`.

Closes #3653

hairyhum commented:
@davidag Yes, I think `data_worker_mod` makes more sense.

davidag commented:
@hairyhum done, let me know if I'm missing something. thanks! 👌🏼

hairyhum commented on implementations/elixir/ockam/ockam/lib/ockam/session/spawner.ex:
```diff
@@ -13,14 +13,14 @@ defmodule Ockam.Session.Spawner do
     defaults to `&default_spawner_setup`
 
-  Upon receiving a message, `worker_mod` worker will be started
+  Upon receiving a message, `data_worker_mod` worker will be started
```
Spawner does not have a handshake and data stages, so this should be just `worker_mod`

davidag replied:
I think it's fixed now 🤔

hairyhum commented:
@davidag Looks good. Can you please address comments in the CLA PR https://github.com/build-trust/ockam-contributors/pull/151 before we can merge the PRs?

davidag commented:
> @davidag Looks good. Can you please address comments in the CLA PR [build-trust/ockam-contributors#151](https://github.com/build-trust/ockam-contributors/pull/151) before we can merge the PRs?

@hairyhum done, also fixed the formatting error and rebased.

hairyhum commented:
Thanks.
@davidag can you also squash your commits, make sure the resulting commit follows the [format](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) and make sure that it's signed by a signature verified by github?

davidag commented:
> Thanks. @davidag can you also squash your commits, make sure the resulting commit follows the [format](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) and make sure that it's signed by a signature verified by github?

@hairyhum Done! 👌🏼

davidag commented:
@hairyhum fixed conflicts with `develop`.
